### ISSUE TITLE
Teachers to be able to assign work to a class or individual - and for this to link to anthologies as discussed

### ISSUE THREAD

**Issue by rebeccacrawford** [ThreeMinuteLearning/my3ml#50]

**Comment by tekul** [comment]
At the moment, stories are organised by the student level. They will be sorted so that the stories matching the student level come first.

My idea was to have a "work queue" concept in addition to this. So a teacher could assign particular stories (or in future, other tasks) by class, or on an individual basis and they would appear first in the student's queue, probably under a different heading (e.g. "My class work"). They could still do other stories first if they wanted to.

**Comment by rebeccacrawford** [comment]
I like the queue idea, given they can ignore it too.

**Comment by tekul** [comment]
I guess if the teacher is assigning the work to the queue then the students shouldn't be ignoring it :). But they can decide that themselves.

**Comment by rebeccacrawford** [comment]
In the old version students couldnt see anything above their own level.

You say below 'student level comes first' but what comes after?, the lower levels they have access to, or can they see all stories?

Can they do stories above their level?

**Comment by tekul** [comment]
Currently they can see all the stories. We can change that if we want - in particular level 0 stories are kind of pointless, so I was going to hide them for other students, and reduce scoring for doing stories which are below your own level. But preventing them from reading  stories that might interest them because they are a different level doesn't seem fair. There are still plenty of interesting things in them.

Presumably the levels are only a rough guide anyway, aside from level 0?

**Comment by rebeccacrawford** [comment]
They are a rough guide (as one P4 kid will be up for a Level 5 story wheareas another will be more comfortable on Level 0 as there is such a range needed.)

Level 0 are really liked by primary teachers who can talk with struggling kids around the image and the simpler sentence. It can lead to good chats. But equally secondary kids have enjoyed them too.

The main thing for the point system is that 3ml is aiming encourage struggling kids as a priority so engagement with the resource not attainment is important.  They shouldn't score any more for harder stories A P6 struggling kid should be able to be at the top of the Leaderboard through effort, not attainment. That is vital and has been successful.

The teacher has been able to control the max level in the past, so as to not put kids off with harder stories than they are ready for. I don think they should be able to see higher stories as if they fail they then give up too easily. The teacher would take this into account, make a Level 2 kid a Level 3 so they can be stretched but also can work at their comfort level.

In the current version kids could work below their level if they wished and shouldn't win less point. This works well. The easy wins encourage the kids along with the leaderboard and the literacy skills are still improved.

Teachers would set a lowish level for many, so they get an easy early experience to see they really can do it, then when they go higher they can still do it even if the words are harder. For others they would let them dive right in at high levels.

**Comment by tekul** [comment]
Yeah, I didn't mean "pointless" other than for higher level kids - I've enjoyed some of the level 0 stories myself and I like the big pictures. But if you want them all to be able to do lower levels that seems Ok.

We can easily filter out higher levels when they are loaded.

**Comment by rebeccacrawford** [comment]
Douglas and I chatted and talked again about how we think students should first see the lower levels through the Starter Stories display. We would like a counter that a teacher can set that specifies the number of stories that a student does before seeing the next level stories.   So for example if the Counter C is 12 then initially the students see 12 level 0 stories irrespective of what level he is.  When he clicks through and completes a story, the next story  that comes up in that tile position is a level 1.  So there is now 11 level 0s and one level 1.  Each time the user clicks through and completes a story that tile shows a level 1, up to the value of C, at which point you get a level 2. This means that they could go straight through to higher levels having only done one level 0 but that's fine as if the teacher wants them to do all the 12 level 0s they can direct them to do that.

**Comment by tekul** [comment]
Some questions

1. What will it be called? Something that explains it's purpose to the user (i.e the teacher). This is logically introducing another "level" for the student (one that they can increment themselves by completing N stories), so this level will also have to be stored for each student in addition to the counter value. It seems like teachers might get confused as to what the actual level is. Would they also be able to see this new level too? 

1. How will this work when the student-teacher mapping isn't one-to-one?

1. Is this an alternative to allowing teachers to set specific stories to come first (the work queue idea) or in addition? It seems to overlap, but doesn't give them much control over what they actually see - just that a number N of some level will come first.

**Comment by rebeccacrawford** [comment]
Answering 2 and 3 first. This is just for the Starter Stories, the stories we think students will do on their own as they are getting the hang of 3ml. At the moment there are 24 tiles. We thought if we used the top 12 for this then the bottom 12 could be for "My work", ie your idea of the teacher being able to set individual tailored queues.

**Comment by rebeccacrawford** [comment]
Answering 1. This is just about creating a starter experience. The student and teachers don't need to know about anything other than their Level. The teachers will tell the students what Level they are.  "You are Level 5, that means you can do stories up to Level 5. You get points for doing all stories but 5 is the highest you can see at the moment. To start off, I would like you to do the Starter Stories on the homepage. You will start with a few Level 0s, then go up as you get the hang of it. Then we can look at other ways of getting stories up to your level."  C will be set to a default, say 6, and in the teacher guidance notes we can tell the teachers they can differentiate the starter experience by showing a higher number of low level starter stories before the students get onto finding stories by searches etc.

**Comment by tekul** [comment]
How the system works for new users is really a separate issue from how it works in general and how teachers can assign items to students who are already familiar with the site. I've created #53 to capture this.

**Comment by rebeccacrawford** [comment]
We chatted about how when the kids do the anthology at the moment it is unfriendly as when they go in and out of individual stories they get taken back to the homepage rather than the anthology view. You had the suggestion that there should be a button by an anthology that allow them to collect all the stories into their work queue, or the teacher would do this.
